### DISCUSSION TITLE
Readonly mode for CalendarFX

### DISCUSSION THREAD

**Discussion by macdevign** [dlemmermann/CalendarFX#144] [Ideas]
Hi,
Like to know 

1)  how do I replace this "Disable Editing" dialog by some other dialog ? What is code that I need to do that ? For example instead of this dialog, I may just replace it by notification message.

<img width="364" alt="image" src="https://user-images.githubusercontent.com/6662383/132136583-e77288bc-3b4c-418c-91fb-9479d43ab661.png">

2) I like to show all calendars in readonly . However it seem disableAll simply just hide those entries of calendars, making those edit either hide all, or show them for edit. So how do I showAll calendars' entries in readonly ?
![image](https://user-images.githubusercontent.com/6662383/132137991-7bcfe7c3-6004-485e-9c4e-0ae7000a8121.png)

thank

**Comment by dlemmermann** [comment]
Browse the source code for the text shown by the dialog, then investigate
how to disable / modify it. If it isn’t possible then add code to
CalendarFX and provide a pull request.

macdevign ***@***.***> schrieb am So. 5. Sept. 2021 um 19:53:

> Hi,
> Like to know how do I replace this "Disable Editing" dialog by some other
> dialog ? What is code that I need to do that ? For example instead of this
> dialog, I may just replace it by notification message.
>
> [image: image]
> <https://user-images.githubusercontent.com/6662383/132136583-e77288bc-3b4c-418c-91fb-9479d43ab661.png>
>
> thank
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dlsc-software-consulting-gmbh/CalendarFX/issues/132>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACIXWXMSY2YUWESBBOHCSFTUAOVBRANCNFSM5DPBBCWQ>
> .
>
-- 
Dirk Lemmermann

Software & Consulting
Zurich, Switzerland
+41-(0)79-800-23-20
http://www.dlsc.com
***@***.***

**Comment by macdevign** [comment]
Hi Dirk,
I have used CalendarFX to build my own project ( aka Calendo just to keep track of completed tasks) and I find that CalendarFX is really useful and extensible. After spending some time on understanding the library code, I able to complete my own project. For this I like to give thank to you for this wonderful library.

However I realize that some answer I need to get a head-start on solving problem is from what been found in on those commented issues, and I have some cool features which I do not think suitable as a patch to the calendarfx but I happy to share those techniques and change-how. (For example, adding tooltip and reusing them for entry cell,   ability to go any view based on mouse position in entry cell, and hiding overlapping month days in Year view (eg currently the default is that day that belong to 1 march is also show on Feb month, which kind of overlapping and hard to see (I unable to find a api to override that behaviour)

Is it possible for you to add Github discussion or any kind of forum so that developers can interact for CalendarFX ? I'm sure this is going to be huge boon to help developers use calenderfx and increase its uptake.

Project Calendo
![image](https://user-images.githubusercontent.com/6662383/133866078-21c8bb50-95c9-4c3f-b5e7-52706db6d3c4.png)

thank

**Comment by dlemmermann** [comment]
I have converted this issue to a discussion. @macdevign feel free to share your ideas here.
